src/backend/optimizer/plan/createplan.c

```diff
@@ -792,10 +783,17 @@ create_join_plan(PlannerInfo *root, JoinPath *best_path)
 		!(root->config->gp_cte_sharing && IsA(plan, HashJoin)))
 	{
 		((Join *) plan)->prefetch_inner = false;
-		((Join *) plan)->prefetch_joinqual = false;
-		((Join *) plan)->prefetch_qual = false;
 	}
 
+	/*
+	 * If we injected a partition selector to the inner side, we must evaluate
+	 * the inner side before the outer side, so that the partition selector
+	 * can influence the execution of the outer side.
+	 */
+	Assert(plan->type == best_path->path.pathtype);
+	if (partition_selector_created)
+		((Join *) plan)->prefetch_inner = true;
```

Comment:
What happens if partition_selector_created in a shared scan?

Reply:
as I understand it, we cannot have a partition selector on the coordinator. and therefore, according to the idea, they should not conflict.